### PR TITLE
Add shapinfos to all machines using coils

### PR DESCRIPTION
**Pull request by blackdragon2447**
Allows viewing the other coils in *EI and rendering in Compass.

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java [hunk 1]
```diff
@@ -990,6 +990,28 @@ public static BiConsumer<ItemStack, List<Component>> createTankTooltips(String n
...
+                    .where('I', ITEM_IMPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('O', ITEM_EXPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('F', FLUID_IMPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('D', FLUID_EXPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('H', MUFFLER_HATCH[GTValues.LV], Direction.SOUTH)
+                    .where('M', MAINTENANCE_HATCH, Direction.NORTH)
+                    .where('#', Blocks.AIR.defaultBlockState());
+                ALL_COILS.entrySet().stream()
+                        .sorted(Comparator.comparingInt(entry -> entry.getKey().getTier()))
+                        .forEach(coil -> shapeInfo.add(builder.where('C', coil.getValue().get()).build()));
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java]:
```suggestion
                        .forEach(coil -> shapeInfo.add(builder.shallowCopy().where('C', coil.getValue().get()).build()));
```

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java [hunk 2]
```diff
@@ -1019,6 +1041,25 @@ public static BiConsumer<ItemStack, List<Component>> createTankTooltips(String n
...
+                    .where('X', CASING_INVAR_HEATPROOF.getDefaultState())
+                    .where('E', ENERGY_INPUT_HATCH[GTValues.LV], Direction.SOUTH)
+                    .where('I', ITEM_IMPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('O', ITEM_EXPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('H', MUFFLER_HATCH[GTValues.LV], Direction.SOUTH)
+                    .where('M', MAINTENANCE_HATCH, Direction.NORTH)
+                    .where('#', Blocks.AIR.defaultBlockState());
+                ALL_COILS.entrySet().stream()
+                        .sorted(Comparator.comparingInt(entry -> entry.getKey().getTier()))
+                        .forEach(coil -> shapeInfo.add(builder.where('C', coil.getValue().get()).build()));
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java]:
```suggestion
                        .forEach(coil -> shapeInfo.add(builder.shallowCopy().where('C', coil.getValue().get()).build()));
```

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java [hunk 3]
```diff
@@ -1048,6 +1089,25 @@ public static BiConsumer<ItemStack, List<Component>> createTankTooltips(String n
...
+                    .where('H', CASING_STAINLESS_CLEAN.getDefaultState()) 
+                    .where('E', ENERGY_INPUT_HATCH[GTValues.LV], Direction.WEST)
+                    .where('I', ITEM_IMPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('F', FLUID_IMPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('D', FLUID_EXPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('M', MAINTENANCE_HATCH, Direction.NORTH)
+                    .where('#', Blocks.AIR.defaultBlockState());
+                ALL_COILS.entrySet().stream()
+                        .sorted(Comparator.comparingInt(entry -> entry.getKey().getTier()))
+                        .forEach(coil -> shapeInfo.add(builder.where('C', coil.getValue().get()).build()));
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTMachines.java]:
```suggestion
                        .forEach(coil -> shapeInfo.add(builder.shallowCopy().where('C', coil.getValue().get()).build()));
```

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 4]
```diff
@@ -411,6 +419,29 @@ public static void init() {}
...
+                    .where('A', Blocks.AIR.defaultBlockState())
+                    .where('E', ENERGY_INPUT_HATCH[GTValues.LV], Direction.SOUTH)
+                    .where('I', ITEM_IMPORT_BUS[GTValues.LV], Direction.WEST)
+                    .where('F', FLUID_IMPORT_HATCH[GTValues.LV], Direction.WEST)
+                    .where('D', FLUID_EXPORT_HATCH[GTValues.LV], Direction.EAST)
+                    .where('H', MUFFLER_HATCH[GTValues.LV], Direction.UP)
+                    .where('M', MAINTENANCE_HATCH, Direction.NORTH);
+                ALL_COILS.entrySet().stream()
+                        .sorted(Comparator.comparingInt(entry -> entry.getKey().getTier()))
+                        .forEach(coil -> shapeInfo.add(builder.where('C', coil.getValue().get()).build()));
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]:
```suggestion
                        .forEach(coil -> shapeInfo.add(builder.shallowCopy().where('C', coil.getValue().get()).build()));
```

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 5]
```diff
@@ -712,6 +743,43 @@ public static void init() {}
...
+                    .where('E', ENERGY_INPUT_HATCH[GTValues.LV], Direction.SOUTH)
+                    .where('L', ITEM_IMPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('O', ITEM_EXPORT_BUS[GTValues.LV], Direction.NORTH)
+                    .where('Q', FLUID_IMPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('D', FLUID_EXPORT_HATCH[GTValues.LV], Direction.NORTH)
+                    .where('H', MUFFLER_HATCH[GTValues.LV], Direction.UP)
+                    .where('M', MAINTENANCE_HATCH, Direction.SOUTH);
+                ALL_COILS.entrySet().stream()
+                        .sorted(Comparator.comparingInt(entry -> entry.getKey().getTier()))
+                        .forEach(coil -> shapeInfo.add(builder.where('C', coil.getValue().get()).build()));
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]:
```suggestion
                        .forEach(coil -> shapeInfo.add(builder.shallowCopy().where('C', coil.getValue().get()).build()));
```

**Review by mikerooni — DISMISSED** [review]
Looks good so far!  
I'd suggest making a shallow copy of the shape builder each time you're modifying it.

**Review by screret — APPROVED** [review]

**Pull request merged**